### PR TITLE
feat(rating): User can rate articles

### PR DESCRIPTION
**Pull request by Princesso**
###  What this PR does
Implements user can rate an article

### Tasks completed
- When a user clicks on the rating stars, it gets the number of ratings
- If user is logged in, it rates successfully and returns average ratings
- If user is not logged in, user is redirected to log in
- Every article being viewed should have an average rating displayed
- Pluralize ratings ie if number of ratings > 1 return users else user

### How to test
- Clone this branch
- Click on an article
- Click on the stars displayed to rate article

### Screenshots

Screenshot 2019-03-13 at 8 21 25 PM
Screenshot 2019-03-13 at 8 23 22 PM

### Relevant pivotal tracker stories
[Finishes #163478763]

### CODE REVIEW

**Review by jsamchineme — APPROVED** [review]
Nice work @Princesso 
LGTM!!!

File: src/components/ArticlePage/index.jsx [hunk 1]
```diff
@@ -9,32 +9,50 @@ import RatingStarsBox from '../../components/RatingStarsBox';
...
+    const { fetchArticle, getArticleAvgRating, match } = this.props;
     let articleId = match.params.articleId;
-    fetchArticle(articleId);
+    await fetchArticle(articleId);
+    getArticleAvgRating(articleId)
+  }
+
+  starClickHandle(item) {
+    let articleId = this.props.match.params.articleId;
+    const _this = this;
```
Comment (ifiokeyo) [inline src/components/ArticlePage/index.jsx]: @Princesso Is there a need to use `line 28` ? Provide context please. No consistency in the use of `this` keyword in this function.

File: src/components/ArticlePage/index.jsx [hunk 2]
```diff
@@ -9,32 +9,50 @@ import RatingStarsBox from '../../components/RatingStarsBox';
...
+      } else if(error.response.data === 'Unauthorized'){
+         toast.error('You need to login to rate an article');
+        return  _this.props.history.push('/login')
+      }
+      toast.error('Error while rating article make sure you are logged in');
+    })
   }
 
   render() {
     const { match, articles = [] } = this.props;
```
Comment (ifiokeyo) [inline src/components/ArticlePage/index.jsx]: @Princesso Consider doing this on `line 43` `const { match: { params: { articleId }, articles = [] } = this.props;`. That way there won't be a need for `line 44`.

File: src/components/ArticlePage/index.jsx [hunk 3]
```diff
@@ -9,32 +9,50 @@ import RatingStarsBox from '../../components/RatingStarsBox';
...
     const { articleId } = match.params;
     const article = articles.find(item => item.id === articleId) || {};
-
     const { 
       title, 
       featuredImage = '/assets/img/map-typing.jpg', 
       author: { username = '', imageUrl: userAvatarImage = '' } = {},
```
Comment (ifiokeyo) [inline src/components/ArticlePage/index.jsx:47]: @Princesso Please explain what `line 49` is meant to do.

**Review by ifiokeyo — CHANGES_REQUESTED** [review]

File: src/components/ArticlePage/index.jsx (hunk 2 above)
Comment (Princesso) [reply src/components/ArticlePage/index.jsx]: @ifiokeyo. Usually, functions that are declared with the function keyword have a different scope. The convention is to use named functions everywhere so in the error handler, the function has a different scope from its parent scope. 

Using unnamed functions however does not change the scope so the this keyword can work with unnamed functions unlike named functions.

File: src/components/ArticlePage/index.jsx (hunk 3 above)
Comment (Princesso) [reply src/components/ArticlePage/index.jsx:47]: I did not work on this section of the single article view page. @jsamchineme  may be able to provide context to what this does.

File: src/components/ArticlePage/index.jsx (hunk 1 above)
Comment (Princesso) [reply src/components/ArticlePage/index.jsx]: @ifiokeyo. Usually, functions that are declared with the function keyword have a different scope. The convention we follow is to use named functions everywhere and using the function keyword makes it easy to create named functions. 

In the error handler, the function keyword was used so it has a different scope from its parent scope.

Using arrow functions however does not change the scope so the this keyword can work with arrow functions unlike functions created with the function keyword.

File: src/components/ArticlePage/index.jsx (hunk 2 above)
Comment (Princesso) [reply src/components/ArticlePage/index.jsx]: I will effect this change and update this PR, thanks.

**Pull request merged**